hibernate-core/src/main/java/org/hibernate/event/internal/DefaultLoadEventListener.java

```diff
@@ -299,6 +299,21 @@ private Object proxyOrLoad(
 					event.getSession()
 			);
 
+			// IMPLEMENTATION NOTE:
+			// The entity's EntityKey may already have been added to the BatchFetchQueue
+			// by ManyToOneType#scheduleBatchLoadIfNeeded.
+			// In the case that the entity is loaded by Session#load or Session#getReference,
+			// it will not be in the BatchFetchQueue, so go ahead and add it.
+			persistenceContext.getBatchFetchQueue().addBatchLoadableEntityKey( keyToLoad );
```

Comment:
I suppose this section should also check for `org.hibernate.persister.entity.EntityPersister#isBatchLoadable` ?

Reply:
Why?  Its done in `#addBatchLoadableEntityKey` - no need to do it twice

Reply:
oh right.

I got confused, but I was checking something odd: it seems like this patch is able to reduce the number of SQL statements being executed even when batching is not enabled.

Trying to figure out why.. ideas?